### DISCUSSION TITLE
Vulkan

### DISCUSSION THREAD

**Discussion by VNNCC** [codam-coding-college/MLX42#55] [Q&A] [answered]
What is the current state with supporting/using Vulkan as rendering backend?

**Comment by W2Wizard** [comment, marked as answer]
Currently I am in the works on converting the library to C++ (17) and switch to support for multiple rendering backends. That is, in the future people can switch and implement any backend they want (OpenGL, Vulkan, ...)

The main reason for this is because in order to get the library accepted on a network wide level, Olivier Crouzet (current head of pedagogy) wants it in Vulkan in order to have it future proof.

Tbh I think its not *really* required as all this library ever will do is draw some simple quads with textures on it. But I think it would still be nice to do it anyway to be a nice intro for students to get into vulkan.

When will it be done? I have no eta yet.
